numba/lowering.py

```diff
@@ -590,8 +594,10 @@ def lower_getitem(self, resty, expr, value, index, signature):
         # Get implementation of getitem
         op = operator.getitem
         fnop = self.context.typing_context.resolve_value_type(op)
-        fnop.get_call_type(self.context.typing_context, signature.args, {})
-        impl = self.context.get_function(fnop, signature)
+        callsig = fnop.get_call_type(
```

Comment:
It is looks like you forget `signature = signature.replace(args=(signature.args[0], types.unliteral(signature.args[1])))` here as it was implemented in first commit to this PR.